thrust/testing/cuda/unique.cu

```diff
@@ -56,13 +46,9 @@ void TestUniqueDevice(ExecutionPolicy exec)
   new_last = new_last_vec[0];
 
   ASSERT_EQUAL(new_last - data.begin(), 7);
-  ASSERT_EQUAL(data[0], 11);
-  ASSERT_EQUAL(data[1], 12);
-  ASSERT_EQUAL(data[2], 20);
-  ASSERT_EQUAL(data[3], 29);
-  ASSERT_EQUAL(data[4], 21);
-  ASSERT_EQUAL(data[5], 31);
-  ASSERT_EQUAL(data[6], 37);
+  data.resize(new_last - data.begin());
```

Comment:
Suggestion: similar to previous refactorings:
```suggestion
  data.erase(new_last, data.end());
```
A few more occurences below.

Reply:
oh so it applies for unique, too. gotcha

Reply:
I didn't even look where `new_last` came from :D But yes! Applies to `unique` as well!

Reply:
also discovered a bug as part of it. fixed